spec/controllers/hyrax/dashboard/collections_controller_spec.rb

```diff
@@ -21,97 +21,133 @@
   let(:unowned_asset)  { create(:work, user: other) }
 
   let(:collection_attrs) do
-    { title: ['My First Collection'], description: ["The Description\r\n\r\nand more"], collection_type_gid: [collection_type_gid] }
+    {
+      title: ['My First Collection'],
+      description: ["The Description\r\n\r\nand more"],
+      collection_type_gid: [collection_type_gid]
+    }
   end
 
+  let(:listener) { Hyrax::Specs::SpyListener.new }
+
+  before { Hyrax.publisher.subscribe(listener) }
+  after  { Hyrax.publisher.unsubscribe(listener) }
+
   describe '#new' do
     before { sign_in user }
 
     it 'assigns @collection' do
       get :new
-      expect(assigns(:collection)).to be_kind_of(Collection)
+      expect(assigns(:collection)).to be_kind_of(Hyrax::PcdmCollection)
     end
   end
 
   describe '#create' do
     before { sign_in user }
 
-    # rubocop:disable RSpec/ExampleLength
-    it "creates a Collection" do
+    it 'creates a Collection with metadata' do # rubocop:disable RSpec/ExampleLength
       expect do
         post :create, params: {
-          collection: collection_attrs.merge(
-            visibility: 'open',
-            # TODO: Tests with old approach to sharing a collection which is deprecated and
-            # will be removed in 3.0.  New approach creates a PermissionTemplate with
-            # source_id = the collection's id.
-            permissions_attributes: [{ type: 'person',
-                                       name: 'archivist1',
-                                       access: 'edit' }]
-          )
+          collection_type_gid: collection_type_gid,
+          pcdm_collection: collection_attrs
         }
       end.to change { Collection.count }.by(1)
-      expect(assigns[:collection].visibility).to eq 'open'
-      expect(assigns[:collection].edit_users).to contain_exactly "archivist1", user.email
+      expect(assigns(:collection)).to be_kind_of(Hyrax::PcdmCollection)
+      expect(assigns[:collection].visibility).to eq 'restricted'
+      expect(assigns[:collection].depositor).to eq user.user_key
+      expect(assigns[:collection].permission_manager.edit_users).to contain_exactly user.user_key
       expect(flash[:notice]).to eq "Collection was successfully created."
     end
 
-    it "removes blank strings from params before creating Collection" do
+    it "publishes collection.metadata.updated" do
       expect do
         post :create, params: {
-          collection: collection_attrs.merge(creator: [''])
-        }
-      end.to change { Collection.count }.by(1)
-      expect(assigns[:collection].title).to eq ["My First Collection"]
-      expect(assigns[:collection].creator).to eq []
-    end
-
-    context "with files I can access" do
```

Comment:
Note: These assets are works and collections that are being added as children of the collection being created.  Is there a use case for this?  You can do this through the edit form or batch process (e.g., I believe this is used on the Dashboard -> Works -> select works -> Add works to collection).  I thought perhaps you could add selected works to a New Collection, but I don't see that as an option through the UI.

Reply:
I think you are correct that you can't select works and create a collection to add them to via the UI (the collection has to already exist) but if Bulkrax is expecting this functionality when importing new works, does that effect this test?